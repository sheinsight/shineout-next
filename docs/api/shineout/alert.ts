const api = JSON.parse('[{"title":"Alert","properties":[{"name":"className","tag":{"cn":"自定义类名","en":"Custom class name","default":"","version":""},"required":false,"type":"string "},{"name":"style","tag":{"cn":"自定义样式","en":"Custom style","default":"","version":""},"required":false,"type":"CSSProperties "}],"cn":"","en":""}]');
export default api;
