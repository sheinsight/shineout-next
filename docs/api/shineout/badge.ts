const api = JSON.parse('[{"title":"Badge","properties":[{"name":"className","tag":{"cn":"自定义类名","en":"Custom class name","default":"","version":""},"required":false,"type":"string "},{"name":"style","tag":{"cn":"自定义样式","en":"Custom style","default":"","version":""},"required":false,"type":"CSSProperties "},{"name":"children","tag":{"cn":"内容","en":"Content","default":"","version":""},"required":false,"type":"ReactNode"},{"name":"color","tag":{"cn":"自定义徽标颜色","en":"Custom badge color","default":"","version":""},"required":false,"type":"string "},{"name":"count","tag":{"cn":"展示的数字，大于 overflowCount 时显示 ${overflowCount}+，为 0 时隐藏","en":"Number to show, show ${overflowCount}+ when it is greater than overflowCount, hide when it is 0","default":"","version":""},"required":false,"type":"ReactNode"},{"name":"dot","tag":{"cn":"小点模式，开启后不展示数字","en":"Dot mode","default":"false","version":""},"required":false,"type":"boolean "},{"name":"offset","tag":{"cn":"偏移量","en":"Offset of the badge","default":"","version":""},"required":false,"type":"[number, number] "},{"name":"overflowCount","tag":{"cn":"封顶数值","en":"Offset of the badge","default":"","version":""},"required":false,"type":"number "},{"name":"showZero","tag":{"cn":"当数值为 0 时，是否展示 Badge","en":"Whether to show Badge when the number is 0","default":"","version":""},"required":false,"type":"boolean "},{"name":"status","tag":{"cn":"徽标状态","en":"Whether to show Badge when the number is 0","default":"","version":""},"required":false,"type":"\\\"success\\\" | \\\"processing\\\" | \\\"default\\\" | \\\"error\\\" | \\\"warning\\\" "},{"name":"text","tag":{"cn":"状态点文本，仅在配置了 status 属性后生效","en":"Status point text, only valid after configuring the status property","default":"","version":""},"required":false,"type":"ReactNode"},{"name":"size","tag":{"cn":"尺寸，仅在非 dot 模式下生效，支持 small 和 default 两种尺寸","en":"Size, only valid in non-dot mode, supports two sizes: small and default","default":"","version":""},"required":false,"type":"\\\"default\\\" | \\\"small\\\" "}],"cn":"","en":"","sort":"0"}]');
export default api;
