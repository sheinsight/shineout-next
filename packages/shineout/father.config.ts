import { defineConfig } from 'father';
export default defineConfig({
  extends: '../../.fatherrc.ts',
});
