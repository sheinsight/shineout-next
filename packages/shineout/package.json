{
  "name": "shineout",
  "version": "3.0.0",
  "description": "A components library for React",
  "keywords": [],
  "license": "ISC",
  "author": "",
  "sideEffects": false,
  "main": "./dist/cjs/index.js",
  "module": "./dist/esm/index.js",
  "typings": "./dist/cjs/index.d.ts",
  "scripts": {
    "build": "father build",
    "build:watch": "father dev",
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "dependencies": {
    "@sheinx/base": "workspace:*",
    "@sheinx/hooks": "workspace:*",
    "@sheinx/shineout-style": "workspace:*"
  },
  "peerDependencies": {
    "classnames": ">=2",
    "immer": ">=10",
    "react": ">=16.8"
  },
  "devDependencies": {
    "jszip": "^3.10.1"
  }
}
