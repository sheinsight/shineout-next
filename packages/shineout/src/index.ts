export { default as Alert } from './alert';
export { default as Button } from './button';
export { default as Checkbox } from './checkbox';
export { default as DatePicker } from './date-picker';
export { default as Dropdown } from './dropdown';
export { default as EditableArea } from './editable-area';
export { default as Form } from './form';
export { default as Icon } from './icon';
export { default as Image } from './image';
export { default as Input } from './input';
export { default as Popover } from './popover';
export { default as Radio } from './radio';
export { default as Rule } from './rule';
export { default as Spin } from './spin';
export { default as Switch } from './switch';
export { default as Tag } from './tag';
export { default as Tabs } from './tabs';
export { default as Textarea } from './textarea';
export { default as Tooltip } from './tooltip';
export { setConfig } from '@sheinx/base';

export * as TYPE from './type';
