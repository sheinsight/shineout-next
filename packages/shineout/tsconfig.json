{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "declaration": true,
    "baseUrl": "./",
    "paths": {
      "shineout": ["./src/index"],
      "@sheinx/base": ["../base/src/index"],
      "@sheinx/hooks": ["../hooks/src/index"]
    }
  },
  "include": ["src"]
}
