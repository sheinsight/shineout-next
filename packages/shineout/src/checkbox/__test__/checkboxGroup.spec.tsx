import React, { useState } from 'react';
import { render, cleanup, screen, fireEvent, waitFor } from '@testing-library/react';
import Checkbox from '..';
import { Form } from 'shineout';
import mountTest from '../../tests/mountTest';
import {
  classTest,
  snapshotTest,
  componentsClassTest,
  textContentTest,
  styleTest,
  classContentTest,
  childrenTest,
  displayTest,
  createClassName,
} from '../../tests/utils';
import CheckboxRawgroup from '../__example__/004-rawgroup';
import CheckboxGroup from '../__example__/005-group';
import CheckboxBlock from '../__example__/006-block';
import CheckboxDisabled from '../__example__/007-disabled-1';
import CheckboxDisabledByFunc from '../__example__/007-disabled-2';

const SO_PREFIX = 'checkbox';
const originClasses = ['wrapper', 'group'];
const originItemClasses = ['wrapperChecked', 'wrapperDisabled', 'groupBlock'];
const {
  wrapper: checkboxClassName,
  wrapperChecked: checkedClassName,
  group: checkboxGroupClassName,
  wrapperDisabled: checkboxDisabledClassName,
  groupBlock: checkboxGroupBlockClassName,
} = createClassName(SO_PREFIX, originClasses, originItemClasses);

const dataRender: string[] = ['red', 'orange', 'yellow', 'green', 'cyan', 'blue', 'violet'];
interface dataObjProps {
  id: number;
  color: string;
}
const dataObj: dataObjProps[] = [
  { id: 1, color: 'red' },
  { id: 2, color: 'cyan' },
  { id: 3, color: 'blue' },
  { id: 4, color: 'green' },
  { id: 5, color: 'yellow' },
  { id: 6, color: 'orange' },
  { id: 7, color: 'violet' },
];

afterEach(cleanup);
describe('CheckboxGroup[Base]', () => {
  mountTest(Checkbox.Group as React.ComponentType);
  childrenTest(Checkbox.Group, checkboxGroupClassName);
  displayTest(Checkbox.Group as React.FC, 'ShineoutCheckboxGroup');
  snapshotTest(<CheckboxRawgroup />, 'by CheckboxRawgroup');
  test('should render structor in CheckboxRawgroup', () => {
    const { container } = render(<CheckboxRawgroup />);
    const checkboxs = container.querySelectorAll(checkboxClassName);

    expect(checkboxs.length).toBe(dataObj.length);
    checkboxs.forEach((checkbox, index) => {
      textContentTest(checkbox, dataObj[index].color);
      if (![3, 5].includes(index + 1)) return;
      classTest(checkbox, checkedClassName);
    });
    fireEvent.click(checkboxs[0]);
    classTest(checkboxs[0], checkedClassName);
    fireEvent.click(checkboxs[2]);
    classTest(checkboxs[2], checkedClassName, false);
  });
  snapshotTest(<CheckboxGroup />, 'by CheckboxGroup');
  test('should render structor in CheckboxGroup', () => {
    const { container } = render(<CheckboxGroup />);
    const checkboxs = container.querySelectorAll(checkboxClassName);
    expect(checkboxs.length).toBe(dataRender.length);
    checkboxs.forEach((checkbox, index) => {
      styleTest(
        screen.getByText(dataRender[index]),
        `border-bottom: 1px solid ${dataRender[index]};`,
      );
      if (!['blue', 'cyan'].includes(dataRender[index])) return;
      classTest(checkbox, checkedClassName);
    });
  });
  snapshotTest(<CheckboxBlock />, 'by CheckboxBlock');
  test('should render when set block', () => {
    const { container } = render(
      <Checkbox.Group
        keygen={(c: any) => c}
        data={dataRender}
        defaultValue={['blue', 'cyan']}
        block
        style={{ marginTop: 32, display: 'inline-block' }}
      />,
    );
    classTest(container.querySelector(checkboxGroupClassName)!, checkboxGroupBlockClassName);
  });
  test('should render when set className and style', () => {
    const { container } = render(
      <Checkbox.Group
        keygen
        data={dataRender}
        className='demo'
        style={{ marginTop: 32, display: 'inline-block' }}
      />,
    );
    const checkboxGroup = container.querySelector(checkboxGroupClassName);
    classTest(checkboxGroup!, 'demo');
    styleTest(checkboxGroup!, 'margin-top: 32px; display: inline-block;');
  });
});
describe('CheckboxGroup[Value]', () => {
  test('should render when set value', () => {
    const selectValue = ['red', 'yellow'];
    const { container } = render(<Checkbox.Group keygen data={dataRender} value={selectValue} />);
    const checkboxs = container.querySelectorAll(checkboxClassName);
    const checkboxsTest = (bool: boolean) => {
      componentsClassTest(checkboxs, bool, checkedClassName, selectValue);
    };
    checkboxsTest(false);
    checkboxsTest(true);
    fireEvent.click(checkboxs[1]);
    checkboxsTest(false);
  });
});
describe('CheckboxGroup[Disabled]', () => {
  snapshotTest(<CheckboxDisabled />, 'by disabled-A');
  snapshotTest(<CheckboxDisabledByFunc />, 'by disabled-B');
  test('should render when set disabled', () => {
    const { container } = render(<CheckboxDisabled />);
    componentsClassTest(
      container.querySelectorAll(checkboxClassName),
      true,
      checkboxDisabledClassName,
    );
  });
  test('should render when set disabled by function', () => {
    const { container } = render(<CheckboxDisabledByFunc />);
    container.querySelectorAll(checkboxClassName).forEach((checkbox) => {
      if (checkbox.textContent !== 'yellow') return;
      classTest(checkbox, checkboxDisabledClassName);
      fireEvent.click(checkbox);
      classTest(checkbox, checkedClassName, false);
    });
  });
  test('should render when set disabled in form and checkboxGroup', () => {
    const changeFn = jest.fn();
    const { container } = render(
      <Form disabled={true}>
        <Checkbox.Group keygen data={dataRender} disabled={false} onChange={changeFn} />
      </Form>,
    );
    fireEvent.click(container.querySelectorAll(checkboxClassName)[0]);
    expect(changeFn.mock.calls.length).toBe(0);
  });
});
describe('Checkbox[Format]', () => {
  const RenderDemo = (props: { format: any }) => {
    const { format } = props;
    const [selected, setSelected] = useState([]);
    return (
      <>
        <Checkbox.Group
          keygen='id'
          data={dataObj}
          value={selected}
          onChange={setSelected}
          format={format}
          renderItem={'color'}
        />
        <div className='render'>{selected}</div>
      </>
    );
  };
  test('should render when set format', () => {
    const RenderTemp = () => {
      const [selected, setSelected] = useState([]);
      return (
        <>
          <Checkbox.Group
            keygen
            data={dataRender}
            value={selected}
            onChange={setSelected}
            format={(e: string) => e + '!'}
          />
          <div className='render'>{selected}</div>
        </>
      );
    };
    const { container } = render(<RenderTemp />);
    fireEvent.click(container.querySelectorAll(checkboxClassName)[2]);
    textContentTest(container.querySelector('.render')!, 'yellow!');
  });
  test('should render when set format by string in obj data', () => {
    const { container } = render(<RenderDemo format='id' />);
    fireEvent.click(container.querySelectorAll(checkboxClassName)[2]);
    textContentTest(container.querySelector('.render')!, '3');
  });
  test('should render when set format value is not in object keys', () => {
    const { container } = render(<RenderDemo format={'name' as 'id'} />);
    fireEvent.click(container.querySelectorAll(checkboxClassName)[2]);
    textContentTest(container.querySelector('.render')!, '');
  });
  test('should render when set format bu function in obj data', () => {
    const { container } = render(<RenderDemo format={(e: dataObjProps) => e.id} />);
    fireEvent.click(container.querySelectorAll(checkboxClassName)[2]);
    textContentTest(container.querySelector('.render')!, '3');
  });
});
describe('Checkbox[RenderItem]', () => {
  test('should render when set renderItem is string', () => {
    const { container } = render(<Checkbox.Group data={dataObj} keygen='id' renderItem='color' />);
    container.querySelectorAll(checkboxClassName).forEach((checkbox, index) => {
      textContentTest(checkbox, dataObj[index].color);
    });
  });
  test('should render when set renderItem is function', () => {
    const { container } = render(
      <Checkbox.Group data={dataObj} keygen='id' renderItem={(e: dataObjProps) => e.color + '!'} />,
    );
    container.querySelectorAll(checkboxClassName).forEach((checkbox, index) => {
      textContentTest(checkbox, dataObj[index].color + '!');
    });
  });
  test('should render when set renderItem is function that return undefined', () => {
    const { container } = render(
      <Checkbox.Group data={dataObj} keygen='id' renderItem={() => undefined} />,
    );
    container.querySelectorAll(checkboxClassName).forEach((checkbox) => {
      textContentTest(checkbox, '');
    });
  });
});
describe('Checkbox[Form]', () => {
  const FormDemo = (valueProps: any) => {
    const [value, setValue] = useState<any>({});
    return (
      <>
        <Form value={value} onChange={setValue}>
          <Form.Item label='demo'>
            <Checkbox.Group
              data={dataObj}
              keygen='id'
              name='selectValue'
              renderItem='color'
              format='id'
              value={valueProps}
            />
          </Form.Item>
        </Form>
        <div className='render'>{value?.selectValue}</div>
      </>
    );
  };
  test('should render in form', async () => {
    const { container } = render(<FormDemo />);
    fireEvent.click(screen.getByText('red'));
    await waitFor(() => {
      textContentTest(container.querySelector('.render')!, '1');
    });
  });
  test('should render in form when set value and name in checkbox', async () => {
    const { container } = render(<FormDemo valueProps={['2', '3']} />);
    const checkboxs =
      container.querySelectorAll(checkboxClassName) ||
      container.querySelectorAll(`.${SO_PREFIX}-wrapper-0-2-16`);
    checkboxs.forEach((checkbox) => {
      classContentTest(checkbox, 'checked', false);
    });
    fireEvent.click(screen.getByText('red'));
    await waitFor(() => {
      textContentTest(container.querySelector('.render')!, '1');
    });
  });
});
describe('Checkbox[Onchange/BeforeChange]', () => {
  test('should render when set onchange and beforeChange', () => {
    const RenderDemo = () => {
      const [selected, setSelected] = useState([]);
      return (
        <>
          <Checkbox.Group
            keygen='id'
            data={dataObj}
            value={selected}
            onChange={setSelected}
            format={'id'}
            renderItem={'color'}
            beforeChange={(e: any) => e + '1'}
          />
          <div className='render'>{selected}</div>
        </>
      );
    };
    const { container } = render(<RenderDemo />);
    fireEvent.click(container.querySelectorAll(checkboxClassName)[2]);
    textContentTest(container.querySelector('.render')!, '31');
  });
});
describe('Checkbox[separator]', () => {
  test('should render when set separator', () => {
    const errorSpy = jest.spyOn(console, 'error').mockImplementation(() => {});
    const RenderDemo = () => {
      const [selected, setSelected] = useState([1]);
      return (
        <>
          <Checkbox.Group
            keygen='id'
            data={dataObj}
            value={selected}
            onChange={setSelected}
            format={'id'}
            renderItem={'color'}
            separator='-'
          />
          <div className='render'>{selected}</div>
        </>
      );
    };
    const { container } = render(<RenderDemo />);
    fireEvent.click(container.querySelectorAll(checkboxClassName)[2]);
    textContentTest(container.querySelector('.render')!, '1-3');
    // error capture
    expect(errorSpy).toHaveBeenCalledWith(
      'use-list-select: separator is string, but value is not string',
    );
  });
});
describe('Checkbox[Prediction]', () => {
  test('should render when set prediction', () => {
    const RenderDemo = () => {
      const [selected, setSelected] = useState<string[] | number[]>(['red']);
      return (
        <>
          <Checkbox.Group
            keygen='id'
            data={dataObj}
            value={selected}
            onChange={setSelected}
            format={'id'}
            renderItem={'color'}
            prediction={(val: string | number, d: dataObjProps) => val === d.color}
          />
          <div className='render'>{selected}</div>
        </>
      );
    };
    const { container } = render(<RenderDemo />);
    const excludes = ['red'];
    const checkboxs = container.querySelectorAll(checkboxClassName);
    const componentsClassTestByBool = () => {
      componentsClassTest(checkboxs, true, checkedClassName, excludes);
      componentsClassTest(checkboxs, !false, checkedClassName, excludes);
    };
    componentsClassTestByBool();
    fireEvent.click(container.querySelectorAll(checkboxClassName)[2]);
    componentsClassTestByBool();
  });
});
