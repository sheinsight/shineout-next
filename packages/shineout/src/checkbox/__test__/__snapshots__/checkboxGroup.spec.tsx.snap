// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxGroup[Base] should render correctly by CheckboxBlock 1`] = `
<div
  class="checkbox-group-0-2-9"
>
  <div
    class="checkbox-wrapper-0-2-1"
  >
    <input
      style="display: none;"
      tabindex="0"
      type="checkbox"
    />
    <div
      class="checkbox-indicatorWrapper-0-2-6"
    >
      <i
        class="checkbox-indicator-0-2-7"
        tabindex="1"
      />
    </div>
    <span
      class="checkbox-desc-0-2-8"
    >
      red
    </span>
  </div>
  <div
    class="checkbox-wrapper-0-2-1"
  >
    <input
      style="display: none;"
      tabindex="0"
      type="checkbox"
    />
    <div
      class="checkbox-indicatorWrapper-0-2-6"
    >
      <i
        class="checkbox-indicator-0-2-7"
        tabindex="1"
      />
    </div>
    <span
      class="checkbox-desc-0-2-8"
    >
      orange
    </span>
  </div>
  <div
    class="checkbox-wrapper-0-2-1"
  >
    <input
      style="display: none;"
      tabindex="0"
      type="checkbox"
    />
    <div
      class="checkbox-indicatorWrapper-0-2-6"
    >
      <i
        class="checkbox-indicator-0-2-7"
        tabindex="1"
      />
    </div>
    <span
      class="checkbox-desc-0-2-8"
    >
      yellow
    </span>
  </div>
  <div
    class="checkbox-wrapper-0-2-1"
  >
    <input
      style="display: none;"
      tabindex="0"
      type="checkbox"
    />
    <div
      class="checkbox-indicatorWrapper-0-2-6"
    >
      <i
        class="checkbox-indicator-0-2-7"
        tabindex="1"
      />
    </div>
    <span
      class="checkbox-desc-0-2-8"
    >
      green
    </span>
  </div>
  <div
    class="checkbox-wrapper-0-2-1 checkbox-wrapperChecked-0-2-3"
  >
    <input
      checked=""
      style="display: none;"
      tabindex="0"
      type="checkbox"
    />
    <div
      class="checkbox-indicatorWrapper-0-2-6"
    >
      <i
        class="checkbox-indicator-0-2-7"
        tabindex="1"
      />
    </div>
    <span
      class="checkbox-desc-0-2-8"
    >
      cyan
    </span>
  </div>
  <div
    class="checkbox-wrapper-0-2-1 checkbox-wrapperChecked-0-2-3"
  >
    <input
      checked=""
      style="display: none;"
      tabindex="0"
      type="checkbox"
    />
    <div
      class="checkbox-indicatorWrapper-0-2-6"
    >
      <i
        class="checkbox-indicator-0-2-7"
        tabindex="1"
      />
    </div>
    <span
      class="checkbox-desc-0-2-8"
    >
      blue
    </span>
  </div>
  <div
    class="checkbox-wrapper-0-2-1"
  >
    <input
      style="display: none;"
      tabindex="0"
      type="checkbox"
    />
    <div
      class="checkbox-indicatorWrapper-0-2-6"
    >
      <i
        class="checkbox-indicator-0-2-7"
        tabindex="1"
      />
    </div>
    <span
      class="checkbox-desc-0-2-8"
    >
      violet
    </span>
  </div>
</div>
`;

exports[`CheckboxGroup[Base] should render correctly by CheckboxGroup 1`] = `
<div
  class="checkbox-group-0-2-9"
>
  <div
    class="checkbox-wrapper-0-2-1"
  >
    <input
      style="display: none;"
      tabindex="0"
      type="checkbox"
    />
    <div
      class="checkbox-indicatorWrapper-0-2-6"
    >
      <i
        class="checkbox-indicator-0-2-7"
        tabindex="1"
      />
    </div>
    <span
      class="checkbox-desc-0-2-8"
    >
      <span
        style="border-bottom: 1px solid red;"
      >
        red
      </span>
    </span>
  </div>
  <div
    class="checkbox-wrapper-0-2-1"
  >
    <input
      style="display: none;"
      tabindex="0"
      type="checkbox"
    />
    <div
      class="checkbox-indicatorWrapper-0-2-6"
    >
      <i
        class="checkbox-indicator-0-2-7"
        tabindex="1"
      />
    </div>
    <span
      class="checkbox-desc-0-2-8"
    >
      <span
        style="border-bottom: 1px solid orange;"
      >
        orange
      </span>
    </span>
  </div>
  <div
    class="checkbox-wrapper-0-2-1"
  >
    <input
      style="display: none;"
      tabindex="0"
      type="checkbox"
    />
    <div
      class="checkbox-indicatorWrapper-0-2-6"
    >
      <i
        class="checkbox-indicator-0-2-7"
        tabindex="1"
      />
    </div>
    <span
      class="checkbox-desc-0-2-8"
    >
      <span
        style="border-bottom: 1px solid yellow;"
      >
        yellow
      </span>
    </span>
  </div>
  <div
    class="checkbox-wrapper-0-2-1"
  >
    <input
      style="display: none;"
      tabindex="0"
      type="checkbox"
    />
    <div
      class="checkbox-indicatorWrapper-0-2-6"
    >
      <i
        class="checkbox-indicator-0-2-7"
        tabindex="1"
      />
    </div>
    <span
      class="checkbox-desc-0-2-8"
    >
      <span
        style="border-bottom: 1px solid green;"
      >
        green
      </span>
    </span>
  </div>
  <div
    class="checkbox-wrapper-0-2-1 checkbox-wrapperChecked-0-2-3"
  >
    <input
      checked=""
      style="display: none;"
      tabindex="0"
      type="checkbox"
    />
    <div
      class="checkbox-indicatorWrapper-0-2-6"
    >
      <i
        class="checkbox-indicator-0-2-7"
        tabindex="1"
      />
    </div>
    <span
      class="checkbox-desc-0-2-8"
    >
      <span
        style="border-bottom: 1px solid cyan;"
      >
        cyan
      </span>
    </span>
  </div>
  <div
    class="checkbox-wrapper-0-2-1 checkbox-wrapperChecked-0-2-3"
  >
    <input
      checked=""
      style="display: none;"
      tabindex="0"
      type="checkbox"
    />
    <div
      class="checkbox-indicatorWrapper-0-2-6"
    >
      <i
        class="checkbox-indicator-0-2-7"
        tabindex="1"
      />
    </div>
    <span
      class="checkbox-desc-0-2-8"
    >
      <span
        style="border-bottom: 1px solid blue;"
      >
        blue
      </span>
    </span>
  </div>
  <div
    class="checkbox-wrapper-0-2-1"
  >
    <input
      style="display: none;"
      tabindex="0"
      type="checkbox"
    />
    <div
      class="checkbox-indicatorWrapper-0-2-6"
    >
      <i
        class="checkbox-indicator-0-2-7"
        tabindex="1"
      />
    </div>
    <span
      class="checkbox-desc-0-2-8"
    >
      <span
        style="border-bottom: 1px solid violet;"
      >
        violet
      </span>
    </span>
  </div>
</div>
`;

exports[`CheckboxGroup[Base] should render correctly by CheckboxRawgroup 1`] = `
<div
  class="checkbox-group-0-2-9"
>
  <div
    class="checkbox-wrapper-0-2-1"
  >
    <input
      style="display: none;"
      tabindex="0"
      type="checkbox"
    />
    <div
      class="checkbox-indicatorWrapper-0-2-6"
    >
      <i
        class="checkbox-indicator-0-2-7"
        tabindex="1"
      />
    </div>
    <span
      class="checkbox-desc-0-2-8"
    >
      red
    </span>
  </div>
  <div
    class="checkbox-wrapper-0-2-1"
  >
    <input
      style="display: none;"
      tabindex="0"
      type="checkbox"
    />
    <div
      class="checkbox-indicatorWrapper-0-2-6"
    >
      <i
        class="checkbox-indicator-0-2-7"
        tabindex="1"
      />
    </div>
    <span
      class="checkbox-desc-0-2-8"
    >
      cyan
    </span>
  </div>
  <div
    class="checkbox-wrapper-0-2-1 checkbox-wrapperChecked-0-2-3"
  >
    <input
      checked=""
      style="display: none;"
      tabindex="0"
      type="checkbox"
    />
    <div
      class="checkbox-indicatorWrapper-0-2-6"
    >
      <i
        class="checkbox-indicator-0-2-7"
        tabindex="1"
      />
    </div>
    <span
      class="checkbox-desc-0-2-8"
    >
      blue
    </span>
  </div>
  <div
    class="checkbox-wrapper-0-2-1"
  >
    <input
      style="display: none;"
      tabindex="0"
      type="checkbox"
    />
    <div
      class="checkbox-indicatorWrapper-0-2-6"
    >
      <i
        class="checkbox-indicator-0-2-7"
        tabindex="1"
      />
    </div>
    <span
      class="checkbox-desc-0-2-8"
    >
      green
    </span>
  </div>
  <div
    class="checkbox-wrapper-0-2-1 checkbox-wrapperChecked-0-2-3"
  >
    <input
      checked=""
      style="display: none;"
      tabindex="0"
      type="checkbox"
    />
    <div
      class="checkbox-indicatorWrapper-0-2-6"
    >
      <i
        class="checkbox-indicator-0-2-7"
        tabindex="1"
      />
    </div>
    <span
      class="checkbox-desc-0-2-8"
    >
      yellow
    </span>
  </div>
  <div
    class="checkbox-wrapper-0-2-1"
  >
    <input
      style="display: none;"
      tabindex="0"
      type="checkbox"
    />
    <div
      class="checkbox-indicatorWrapper-0-2-6"
    >
      <i
        class="checkbox-indicator-0-2-7"
        tabindex="1"
      />
    </div>
    <span
      class="checkbox-desc-0-2-8"
    >
      orange
    </span>
  </div>
  <div
    class="checkbox-wrapper-0-2-1"
  >
    <input
      style="display: none;"
      tabindex="0"
      type="checkbox"
    />
    <div
      class="checkbox-indicatorWrapper-0-2-6"
    >
      <i
        class="checkbox-indicator-0-2-7"
        tabindex="1"
      />
    </div>
    <span
      class="checkbox-desc-0-2-8"
    >
      violet
    </span>
  </div>
</div>
`;

exports[`CheckboxGroup[Disabled] should render correctly by disabled-A 1`] = `
<div
  class="checkbox-group-0-2-9"
>
  <div
    class="checkbox-wrapper-0-2-1 checkbox-wrapperDisabled-0-2-5"
  >
    <input
      disabled=""
      style="display: none;"
      tabindex="0"
      type="checkbox"
    />
    <div
      class="checkbox-indicatorWrapper-0-2-6"
    >
      <i
        class="checkbox-indicator-0-2-7"
        tabindex="-1"
      />
    </div>
    <span
      class="checkbox-desc-0-2-8"
    >
      red
    </span>
  </div>
  <div
    class="checkbox-wrapper-0-2-1 checkbox-wrapperDisabled-0-2-5"
  >
    <input
      disabled=""
      style="display: none;"
      tabindex="0"
      type="checkbox"
    />
    <div
      class="checkbox-indicatorWrapper-0-2-6"
    >
      <i
        class="checkbox-indicator-0-2-7"
        tabindex="-1"
      />
    </div>
    <span
      class="checkbox-desc-0-2-8"
    >
      orange
    </span>
  </div>
  <div
    class="checkbox-wrapper-0-2-1 checkbox-wrapperDisabled-0-2-5"
  >
    <input
      disabled=""
      style="display: none;"
      tabindex="0"
      type="checkbox"
    />
    <div
      class="checkbox-indicatorWrapper-0-2-6"
    >
      <i
        class="checkbox-indicator-0-2-7"
        tabindex="-1"
      />
    </div>
    <span
      class="checkbox-desc-0-2-8"
    >
      yellow
    </span>
  </div>
  <div
    class="checkbox-wrapper-0-2-1 checkbox-wrapperDisabled-0-2-5"
  >
    <input
      disabled=""
      style="display: none;"
      tabindex="0"
      type="checkbox"
    />
    <div
      class="checkbox-indicatorWrapper-0-2-6"
    >
      <i
        class="checkbox-indicator-0-2-7"
        tabindex="-1"
      />
    </div>
    <span
      class="checkbox-desc-0-2-8"
    >
      green
    </span>
  </div>
  <div
    class="checkbox-wrapper-0-2-1 checkbox-wrapperDisabled-0-2-5 checkbox-wrapperChecked-0-2-3"
  >
    <input
      checked=""
      disabled=""
      style="display: none;"
      tabindex="0"
      type="checkbox"
    />
    <div
      class="checkbox-indicatorWrapper-0-2-6"
    >
      <i
        class="checkbox-indicator-0-2-7"
        tabindex="-1"
      />
    </div>
    <span
      class="checkbox-desc-0-2-8"
    >
      cyan
    </span>
  </div>
  <div
    class="checkbox-wrapper-0-2-1 checkbox-wrapperDisabled-0-2-5 checkbox-wrapperChecked-0-2-3"
  >
    <input
      checked=""
      disabled=""
      style="display: none;"
      tabindex="0"
      type="checkbox"
    />
    <div
      class="checkbox-indicatorWrapper-0-2-6"
    >
      <i
        class="checkbox-indicator-0-2-7"
        tabindex="-1"
      />
    </div>
    <span
      class="checkbox-desc-0-2-8"
    >
      blue
    </span>
  </div>
  <div
    class="checkbox-wrapper-0-2-1 checkbox-wrapperDisabled-0-2-5"
  >
    <input
      disabled=""
      style="display: none;"
      tabindex="0"
      type="checkbox"
    />
    <div
      class="checkbox-indicatorWrapper-0-2-6"
    >
      <i
        class="checkbox-indicator-0-2-7"
        tabindex="-1"
      />
    </div>
    <span
      class="checkbox-desc-0-2-8"
    >
      violet
    </span>
  </div>
</div>
`;

exports[`CheckboxGroup[Disabled] should render correctly by disabled-B 1`] = `
<div
  class="checkbox-group-0-2-9"
>
  <div
    class="checkbox-wrapper-0-2-1"
  >
    <input
      style="display: none;"
      tabindex="0"
      type="checkbox"
    />
    <div
      class="checkbox-indicatorWrapper-0-2-6"
    >
      <i
        class="checkbox-indicator-0-2-7"
        tabindex="1"
      />
    </div>
    <span
      class="checkbox-desc-0-2-8"
    >
      red
    </span>
  </div>
  <div
    class="checkbox-wrapper-0-2-1"
  >
    <input
      style="display: none;"
      tabindex="0"
      type="checkbox"
    />
    <div
      class="checkbox-indicatorWrapper-0-2-6"
    >
      <i
        class="checkbox-indicator-0-2-7"
        tabindex="1"
      />
    </div>
    <span
      class="checkbox-desc-0-2-8"
    >
      orange
    </span>
  </div>
  <div
    class="checkbox-wrapper-0-2-1 checkbox-wrapperDisabled-0-2-5"
  >
    <input
      disabled=""
      style="display: none;"
      tabindex="0"
      type="checkbox"
    />
    <div
      class="checkbox-indicatorWrapper-0-2-6"
    >
      <i
        class="checkbox-indicator-0-2-7"
        tabindex="-1"
      />
    </div>
    <span
      class="checkbox-desc-0-2-8"
    >
      yellow
    </span>
  </div>
  <div
    class="checkbox-wrapper-0-2-1"
  >
    <input
      style="display: none;"
      tabindex="0"
      type="checkbox"
    />
    <div
      class="checkbox-indicatorWrapper-0-2-6"
    >
      <i
        class="checkbox-indicator-0-2-7"
        tabindex="1"
      />
    </div>
    <span
      class="checkbox-desc-0-2-8"
    >
      green
    </span>
  </div>
  <div
    class="checkbox-wrapper-0-2-1 checkbox-wrapperChecked-0-2-3"
  >
    <input
      checked=""
      style="display: none;"
      tabindex="0"
      type="checkbox"
    />
    <div
      class="checkbox-indicatorWrapper-0-2-6"
    >
      <i
        class="checkbox-indicator-0-2-7"
        tabindex="1"
      />
    </div>
    <span
      class="checkbox-desc-0-2-8"
    >
      cyan
    </span>
  </div>
  <div
    class="checkbox-wrapper-0-2-1 checkbox-wrapperChecked-0-2-3"
  >
    <input
      checked=""
      style="display: none;"
      tabindex="0"
      type="checkbox"
    />
    <div
      class="checkbox-indicatorWrapper-0-2-6"
    >
      <i
        class="checkbox-indicator-0-2-7"
        tabindex="1"
      />
    </div>
    <span
      class="checkbox-desc-0-2-8"
    >
      blue
    </span>
  </div>
  <div
    class="checkbox-wrapper-0-2-1"
  >
    <input
      style="display: none;"
      tabindex="0"
      type="checkbox"
    />
    <div
      class="checkbox-indicatorWrapper-0-2-6"
    >
      <i
        class="checkbox-indicator-0-2-7"
        tabindex="1"
      />
    </div>
    <span
      class="checkbox-desc-0-2-8"
    >
      violet
    </span>
  </div>
</div>
`;
